### PR TITLE
only enable in sql Org source block

### PR DESCRIPTION
**Pull request by stardiviner**
Currently, ejc-sql enabled in all Org source blocks. I think it should only be
enabled in sql source blocks.

### CODE REVIEW

**Comment by stardiviner** [comment]
@kostafey Hi, what about this PR? You might missed it, so I ping here.

**Comment by kostafey** [comment]
@stardiviner ok, first of all, `=` function in Elisp is only sutable for numbers or markers. You can't use it to compare strings, so please use `equal` function. Furtermore, `org-src--babel-info` variable is not initialized in the time of `org-src-mode-hook` execution. So, please update your pull request. You can consider the following diff. Looks like, it works as extected.

```diff
diff --git a/ejc-sql.el b/ejc-sql.el
index e1a5b76..2210b9a 100644
--- a/ejc-sql.el
+++ b/ejc-sql.el
@@ -359,6 +359,21 @@ Prepare buffer to operate as `ejc-sql-mode' buffer."
           (insert-file-contents (ejc-get-result-file-path))
           (buffer-string)))))
 
+(defun ejc-get-product-name (db)
+  (or (alist-get :subprotocol db)
+      (alist-get :dbtype db)))
+
+(defun ejc-org-edit-special (orig-fun &rest args)
+  (if (and (equal "sql" (car (org-babel-get-src-block-info)))
+           (boundp 'ejc-db) ejc-db
+           (boundp 'ejc-connection-name) ejc-connection-name)
+      (let* ((db ejc-db)
+             (connection-name ejc-connection-name)
+             (product-name (ejc-get-product-name db)))
+        (apply orig-fun args)
+        (ejc-add-connection product-name connection-name db))
+    (apply orig-fun args)))
+
 ;;;###autoload
 (defun ejc-connect (connection-name)
   "Connect to selected db."
@@ -373,14 +388,12 @@ Prepare buffer to operate as `ejc-sql-mode' buffer."
                    (or (lax-plist-get conn-statistics c2) 0)))
               conn-list)))))
   (let* ((db (cdr (ejc-find-connection connection-name)))
-         (product-name (or (alist-get :subprotocol db)
-                           (alist-get :dbtype db))))
+         (product-name (ejc-get-product-name db)))
     (ejc-update-conn-statistics connection-name)
     (ejc-configure-sql-buffer product-name)
     (when (derived-mode-p 'org-mode)
       (defalias 'org-babel-execute:sql 'ejc-eval-org-snippet)
-      (add-hook 'org-src-mode-hook
-                (lambda () (ejc-add-connection product-name connection-name db))))
+      (advice-add 'org-edit-special :around #'ejc-org-edit-special))
     (setq-local ejc-connection-name connection-name)
     (setq-local ejc-db db)
     (message "Connection started...")
```

**Comment by stardiviner** [comment]
Your advice is better. I have re-created commit.

File: ejc-sql.el [hunk 1]
```diff
@@ -373,14 +388,12 @@ Prepare buffer to operate as `ejc-sql-mode' buffer."
                    (or (lax-plist-get conn-statistics c2) 0)))
               conn-list)))))
   (let* ((db (cdr (ejc-find-connection connection-name)))
-         (product-name (or (alist-get :subprotocol db)
-                           (alist-get :dbtype db))))
+         (product-name (ejc-get-product-name)))
```
Comment (kostafey) [inline ejc-sql.el]: Pass param `db` to `ejc-get-product-name` fn.

**Comment by kostafey** [comment]
Please, update your pull request: pass param `db` to `ejc-get-product-name` fn, make sure all tests passed.

**Comment by kostafey** [comment]
The main advantage of h2 database - you don't need to install it. You need only Java and h2 JDBC driver (basically in your .m2 directory). It should be automatically installed (if no yet) by maven during `make test` run.
If not, please check your .m2 folder manually. Furthermore, check `.m2/settings.xml` file (looks like something unexpected there).

To update cask dependencies you should run `make elpa`. Actually, the test scenario includes this run, see:
https://github.com/kostafey/ejc-sql/blob/master/.travis.yml#L23

**Comment by stardiviner** [comment]
I installed H2 library through maven manually now. (I checked the `ejc-tests.el` file, found the `(ejc-test:run-maven-dependency-plugin)`, but still got the failed test:
```
Ran 8 tests containing 28 assertions.
0 failures, 0 errors.
"

   passed  3/3  ejc-test:run-lein-test

Ran 3 tests, 2 results as expected, 1 unexpected (2019-01-15 08:59:15+0800)

1 unexpected results:
   FAILED  ejc-test:get-connection

make: *** [/home/stardiviner/Code/Emacs/ejc-sql/Makefile:5: test] Error 1
make: Leaving directory '/home/stardiviner/Code/Emacs/ejc-sql'
```

Anyway, thanks for your code and merging this patch.

**Pull request merged**